src/CSharp/CodeCracker/Design/InconsistentAccessibility/InconsistentAccessibilityCodeFixProvider.cs

```diff
@@ -0,0 +1,169 @@
+using Microsoft.CodeAnalysis.Text;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace CodeCracker.CSharp.Design.InconsistentAccessibility
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = nameof(InconsistentAccessibilityCodeFixProvider)), Shared]
+    public sealed class InconsistentAccessibilityCodeFixProvider : CodeFixProvider
+    {
+        internal const string InconsistentAccessibilityInMethodParameterCompilerErrorNumber = "CS0051";
+
+        public override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(InconsistentAccessibilityInMethodParameterCompilerErrorNumber);
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var diagnostic = context.Diagnostics.First();
+
+            var inconsistentAccessibilityInfo = await GetInconsistentAccessibilityInfoAsync(context.Document, diagnostic, context.CancellationToken).ConfigureAwait(false);
+
+            var typeLocation = await FindTypeLocationInSourceCodeAsync(context.Document, inconsistentAccessibilityInfo.TypeToChangeAccessibility, context.CancellationToken).ConfigureAwait(false);
+
+            if(typeLocation != Location.None)
+            {
+                context.RegisterCodeFix(CodeAction.Create(inconsistentAccessibilityInfo.CodeActionMessage, c => ChangeTypeAccessibilityAsync(context.Document.Project.Solution, inconsistentAccessibilityInfo.NewAccessibilityModifiers, typeLocation, c)), diagnostic);
+            }
+        }
+
+        private static async Task<InconsistentAccessibilityInfo> GetInconsistentAccessibilityInfoAsync(Document document, Diagnostic diagnostic, CancellationToken cancellationToken)
+        {
+            InconsistentAccessibilityInfoProvider inconsistentAccessibilityProvider = null;
+
+            switch (diagnostic.Id)
+            {
+                case InconsistentAccessibilityInMethodParameterCompilerErrorNumber:
+                    inconsistentAccessibilityProvider = new InconsistentAccessibilityInMethodParameter();
+                    break;
+            }
+
+            return await inconsistentAccessibilityProvider.GetInconsistentAccessibilityInfoAsync(document, diagnostic, cancellationToken).ConfigureAwait(false);
+        }
+
+        private static async Task<Location> FindTypeLocationInSourceCodeAsync(Document document, TypeSyntax type, CancellationToken cancellationToken)
+        {
+            var result = Location.None;
+
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var typeSymbol = semanticModel.GetSymbolInfo(type, cancellationToken).Symbol;
+
+            if(typeSymbol != null)
+            {
+                var locationInSource = typeSymbol.Locations.FirstOrDefault(location => location.IsInSource);
```

Comment:
You may find more than one location for a type if the type is declared `partial`. I suggest you return a `Task<Location[]>` or something similar.